packages/components/src/viewport/withPlaceholder/index.js

```diff
@@ -0,0 +1,27 @@
+'use strict';
+
+import React from 'react';
+import { View } from 'react-native';
+
+export default WrappedComponent => {
+  return class extends React.Component {
+    constructor(props, context) {
+      super(props, context);
+    }
+
+    render() {
+      if (this.props.inViewport) {
+        return <WrappedComponent {...this.props} />;
+      }
+      return <View style={this.props.placeholderStyle} />;
+    }
+
+    static propTypes = {
```

Comment:
What about proptypes of the wrapped component?

Reply:
I think that the HOC should not validate the `propTypes` of the wrapped component since it only passes them down to it. This validation can lead to problems.

For example, because of this https://github.com/netceteragroup/girders-elements/commit/2e34011c3ff513e98a3a93d2c630d2ec783e6e7e#diff-7622c24a1e605ed50ad9ebd7f0b61699R57, we get an unnecessary warning: ```Warning: Failed prop type: The prop `inViewport` is marked as required in `ViewportAware(WithPlaceholder(Image))`, but its value is `undefined`.```

The `Viewport.Aware` component requires the very property (`inViewport`) that it has to set for the wrapped component (in this case, `WithPlaceholder`).